### PR TITLE
enable referrals

### PR DESCRIPTION
**Pull request by anene**

### CODE REVIEW

File: lib/install-redirect.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { SOURCE_TO_INSTALL_LINK_REGEX, SOURCE_PARAM_STORAGE, INSTALL_LINK } from "../cfg/sources.js";
+
+for (const [platform, regex] of Object.entries( SOURCE_TO_INSTALL_LINK_REGEX )) {
```
Comment (cbonnoit) [inline lib/install-redirect.js]: "source" or "platform"?

File: cfg/sources.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+export const SOURCES = Object.freeze({
```
Comment (cbonnoit) [inline cfg/sources.js]: `source` sounds a bit too generic to me - is there somethign slightly narrower we could get happy with? i worry (a) there are other things in our infra that also touch `cbonnoit.github.io` that also have a sense of a source (I'm thinking of a `TranscriptSource` and `AudioSource`) and (b) even for attribution, you might be able to think up multiple distinct attributes around sourcing, while i think with this attribute you really only intend to be enumerating the 'channel' in a marketing sense through which the users arrived.

anyway, how about `traffic_channel`? or do you have any other ideas?

File: cfg/sources.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    [SOURCES.LINKEDIN_DM]: new RegExp('get-trellus', 'i'),
+    [SOURCES.LINKEDIN_POST]: new RegExp('try-trellus', 'i'),
+    [SOURCES.TRELLUS_WEBSITE]: new RegExp('get-started', 'i'),
+    [SOURCES.MANUAL]: new RegExp('install-now', 'i')
+  });
+
+
+export const SOURCE_PARAM_STORAGE = 'source'
+export const REFERRER_PARAM_STORAGE = 'referrer'
+export const REFERRER_PARAM_URL = 'ref'
```
Comment (cbonnoit) [inline cfg/sources.js]: can we get some short comments here on what these are? (e.g. some are local storage keys , some are query params that will be looked for, in all cases it'd be nice to know what sort of object would be associated with the key / param)

File: pages/login/index.js [hunk 4]
```diff
@@ -120,7 +147,7 @@ function setApiKey(apiKey) {
   if (result == null)
     addBanner('Service worker did not acknowledge', true)
   else if (result === true || result['success'] === true)
-    onSuccessRedirectBasedOfPageWidth()
+    submitTrackerInformationAndRedirect(apiKey)
```
Comment (cbonnoit) [inline pages/login/index.js]: this is fine but if you want your functions to be slightly more atomic this seems like a natural candidate for promise chaining, e.g. you can get the same thing from `submitTrackerInformation().then(() => redirectBasedOfPageWidth())` but then your functions each do one thing

File: cfg/sources.js (hunk 2 above)
Comment (anene) [reply cfg/sources.js]: called attribution_channel

File: lib/install-redirect.js (hunk 1 above)
Comment (anene) [reply lib/install-redirect.js]: renamed

**Pull request merged**